sdk/messaging/azeventhubs/consumer_client.go

```diff
@@ -106,32 +106,33 @@ func NewConsumerClient(consumerGroup string, fullyQualifiedNamespace string, eve
 	}, options)
 }
 
-// NewConsumerClientFromConnectionString creates a ConsumerClient with a connection string that contains an EntityPath value.
+// NewConsumerClientFromConnectionString creates a ConsumerClient with a connection string that does not have an entity path value.
 // The consumerGroup is the consumer group for this consumer.
+//
+// connectionString can be one of the following formats:
+//
+// Connection string, no EntityPath. In this case eventHub cannot be empty.
+// ex: Endpoint=sb://<your-namespace>.servicebus.windows.net/;SharedAccessKeyName=<key-name>;SharedAccessKey=<key>
+//
+// Connection string, has EntityPath. In this case eventHub must be empty.
 // ex: Endpoint=sb://<your-namespace>.servicebus.windows.net/;SharedAccessKeyName=<key-name>;SharedAccessKey=<key>;EntityPath=<entity path>
-func NewConsumerClientFromConnectionString(consumerGroup string, connectionString string, partitionID string, options *ConsumerClientOptions) (*ConsumerClient, error) {
+func NewConsumerClientFromConnectionString(connectionString string, eventHub string, partitionID string, consumerGroup string, options *ConsumerClientOptions) (*ConsumerClient, error) {
```

Comment:
Shouldn't this be.
```go
func NewConsumerClientFromConnectionString(eventHub string, partitionID string, consumerGroup string, connectionString string, options *ConsumerClientOptions) (*ConsumerClient, error)
```

Reply:
`(<RP specific params>, <cred param>, <options bag>)`

Reply:
In this particular case the connection string is the endpoint. I can move it.

Reply:
Leave as first.

Reply:
Talked with @jhendrixMSFT , we'll leave it for now. Looks like we did this in the blob client, and led with the connection string.

In this particular case the connection string contains both an endpoint and a key, so it can make sense in either position.